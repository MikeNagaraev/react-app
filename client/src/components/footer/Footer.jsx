import React from "react";

export default class Footer extends React.Component {
    constructor(props) {
        super(props);
        this.props = props;
    }
    
    render () {
        return (
            <footer className="footer container">
                <p>{this.props.text}</p>
            </footer>
        )
    }
}
