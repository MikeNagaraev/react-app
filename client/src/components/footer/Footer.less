@footer-height: 5rem;
.footer {
    height: @footer-height;
    background: @black2;
    &>p {
        color: @red1;
        font-size: 1.5rem;
        line-height: @footer-height;
        padding-left: 7rem;
    }
}