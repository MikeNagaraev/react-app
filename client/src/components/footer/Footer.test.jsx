import * as React from 'react';
import { mount, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import renderer from 'react-test-renderer';

import Footer from './Footer';

describe('Footer component', () => {
    beforeAll(() => {
        configure({ adapter: new Adapter() });
    });

    it('should render component', () => {
        const enzymeWrapper = mount(<Footer />);
        expect(enzymeWrapper.find('footer')).toHaveLength(1);
    });

    it('renders correctly snapshot', () => {
        const tree = renderer.create(
            <Footer text="netflixroulette"/>
        ).toJSON();
        expect(tree).toMatchSnapshot();
    });
});
