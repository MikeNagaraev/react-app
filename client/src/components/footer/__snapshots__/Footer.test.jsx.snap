// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer component renders correctly snapshot 1`] = `
<footer
  className="footer container"
>
  <p>
    netflixroulette
  </p>
</footer>
`;
